uvicorn/protocols/websockets/wsproto_impl.py

```diff
@@ -248,7 +249,15 @@ async def send(self, message):
                 )
                 self.handshake_complete = True
                 subprotocol = message.get("subprotocol")
-                extra_headers = message.get("headers", [])
+                headers = list(message.get("headers", [])) + self.default_headers
+                extra_headers = []
+                _added_names = []
+                for name, value in headers:
+                    if name.lower() in _added_names:
+                        continue
+                    _added_names.append(name.lower())
+                    extra_headers.append((name, value))
+
```

Comment:
Why not just this? 🤔 
```suggestion
                extra_headers = list(message.get("headers", [])) + self.default_headers
```

Reply:
if the user passes custom Server header, it lead to an exception of duplicate header names. That's why added a deduplication logic

Reply:
What exception?

Reply:
In `websockets` implementation 

```python
  File "D:\ProgramData\Anaconda3\envs\general\lib\site-packages\websockets\datastructures.py", line 119, in __getitem__
    raise MultipleValuesError(key)
websockets.datastructures.MultipleValuesError: 'Server'
```

In `wsproto` we get duplicate `Server` headers.

Reply:
This should be a bug on `websockets` side. 🤔

It's ok to have multiple headers with the same key, it's the behavior we have on `http` protocols.

Reply:
Oh, okay! will do the necessary changes. And we should add that custom `Server` header is not allowed for `websockets`.

Reply:
Let's not allow on `websockets`.